managed/models/settings_helpers.go

```diff
@@ -433,6 +440,18 @@ func ValidateSettings(params *ChangeSettingsParams) error {
 		return errors.New("both pmm_public_address and remove_pmm_public_address are present")
 	}
 
+	if params.DefaultRoleID != 0 {
+		tx, ok := q.(*reform.TX)
+		if !ok {
+			return errors.New("changing Role ID requires a *reform.TX")
+		}
+
+		var r Role
+		if err := FindAndLockRole(tx, params.DefaultRoleID, &r); err != nil {
+			return err
+		}
```

Comment:
Why do you think it does not represent validation anymore?
The new piece of code validates if the role exists and if it can be changed. Isn't this expected from settings validation?

Reply:
It's locking some entity here, that's not something I would expect from validation function. This function was more about validation for correctness, like `email` address has valid format, not like `email` address exists.

I think we can extract `FindAndLock` in UpdateSettings because it's more about DB consistency rather than parameters validation

Reply:
Updated